distage/distage-framework-docker/src/main/scala/izumi/distage/docker/DockerContainer.scala

```diff
@@ -245,6 +247,26 @@ object DockerContainer {
             case Right(mappedPorts) =>
               val container = DockerContainer[T](ContainerId(res.getId), inspection.getName, mappedPorts, config, clientw.clientConfig, Map.empty)
               logger.debug(s"Created $container from ${config.image}...")
+
+              if (config.networks.nonEmpty) {
+                logger.debug(s"Going to attach container ${res.getId -> "id"} to ${config.networks -> "networks"}")
+
+                val existedNetworks: List[Network] = client.listNetworksCmd().exec().asScala.toList
+
+                config.networks.map {
+                  network =>
+                    existedNetworks.find(_.getName == network.name).fold {
```

Comment:
Maybe we should create networks with random name and manage reuse same as we do with containers? /cc @pshirshov

Reply:
@neko-kai I've thought of creation resource managed networks. To bind them via weak sets. But there was trouble that in such case we will need to add network binding, add a dependency to the container (to strengthen reference, I thought of method like `dependsOnDocker`) and modify container config to get an updated list of networks. Which is a little bit overcomplicated then just add networks in one place.

Reply:
>  modify container config to get an updated list of networks
Only if we would have any chance to do this during `dependsOnNetwork` method (like add a dependency to `ProvideMagnet` and modify config before instantiation.)